lib/internal/test_runner/test.js

```diff
@@ -409,17 +409,26 @@ class Test extends AsyncResource {
   }
 
   matchesTestNamePatterns() {
-    const matchesByNameOrParent = ArrayPrototypeSome(testNamePatterns, (re) =>
+    let patterns;
+    let boolOp = 0;
```

Comment:
Why is this called boolOp and it's 0/1 based (and not a boolean)?

Reply:
I figured that number was easier for me to work with, if Boolean is better I can fix it

Reply:
Yes, please use booleans for booleans and numbers for numbers. It'd be a lot easier to just call it `isMatchingOrSkippingTests` (and probably rename the function to something like `filterTestsByName`?)

Reply:
I can do that, just to want to make sure renaming the function won't break another file